File: frontend/src/api/workspace.ts
import request from '@/utils/request'

export interface Workspace {
  id: number
  name: string
  ownerId: number
  createdAt?: string
  updatedAt?: string
}

export function listWorkspaces() {
  return request.get<any, { data: Workspace[] }>('/workspaces')
}

export function getWorkspace(id: number) {
  return request.get<any, { data: Workspace }>(`/workspaces/${id}`)
}

File: backend/src/main/java/com/fast/knowledge/model/entity/Workspace.java
package com.fast.knowledge.model.entity;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Workspace {
    private Long id;
    private String name;
    private Long ownerId;
    private String settings;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
